js/Utils/Utils.js

```diff
@@ -86,6 +86,32 @@ SPM.Utils = {
 
         // pass in the target node, as well as the observer options
         observer.observe(target, config);
+    },
+
+    removeFromArray: function(key, array) {
+        var index = array.indexOf(key);
+        if (index > -1) {
+            array.splice(index, 1);
+            return true;
+        }
+        return false;
+    },
+
+    removeFromObject: function(key, object) {
```

Comment:
If I'm not mistaken, that's exactly what `delete` does. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/delete